ndarray-rand/src/lib.rs

```diff
@@ -6,33 +6,50 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! Constructors for randomized arrays. `rand` integration for `ndarray`.
+//! Constructors for randomized arrays: `rand` integration for `ndarray`.
 //!
 //! See [**`RandomExt`**](trait.RandomExt.html) for usage examples.
 //!
-//! **Note:** `ndarray-rand` depends on `rand` 0.7. If you use any other items
-//! from `rand`, you need to specify a compatible version of `rand` in your
-//! `Cargo.toml`. If you want to use a RNG or distribution from another crate
-//! with `ndarray-rand`, you need to make sure that crate also depends on the
-//! correct version of `rand`. Otherwise, the compiler will return errors
-//! saying that the items are not compatible (e.g. that a type doesn't
-//! implement a necessary trait).
+//! ## Note
+//!
+//! `ndarray-rand` depends on [`rand` 0.7.0](https://docs.rs/rand/0.7.0/rand/).
+//!
+//! [`rand`](https://docs.rs/rand/0.7.0/rand/) and [`rand-distr`](https://docs.rs/rand_distr/0.2.1/rand_distr/)
+//! are re-exported as sub-modules, [`ndarray_rand::rand`](rand/index.html)
+//! and [`ndarray_rand::rand_distr`](rand_distr/index.html) respectively.
+//! Please rely on these submodules for guaranteed version compatibility.
+//!
+//! If you want to use a random number generator or distribution from another crate
+//! with `ndarray-rand`, you need to make sure that the other crate also depends on the
+//! same version of `rand`. Otherwise, the compiler will return errors saying
+//! that the items are not compatible (e.g. that a type doesn't implement a
+//! necessary trait).
 
-use rand::distributions::Distribution;
-use rand::rngs::SmallRng;
-use rand::{thread_rng, Rng, SeedableRng};
+use crate::rand::distributions::Distribution;
+use crate::rand::rngs::SmallRng;
+use crate::rand::{thread_rng, Rng, SeedableRng};
 
 use ndarray::ShapeBuilder;
 use ndarray::{ArrayBase, DataOwned, Dimension};
 
+/// [`rand`](https://docs.rs/rand/0.7.0/rand/), re-exported for convenience and version-compatibility.
+pub mod rand {
```

Comment:
Might be my ignorance, I thought we had pub extern crate for re-exporting, but I'm unsure what difference it makes?

Reply:
Never heard of `pub extern crate <...>`! 
Digging around, it seems they have exactly the same effect ( https://github.com/nix-rust/nix/pull/694 ). 
Using `mod` has the plus of being more explicit in my eyes, given that it uses the "standard"  visibility mechanism of the 2018 edition.